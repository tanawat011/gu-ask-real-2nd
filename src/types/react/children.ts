export type Children = { children: JSX.Element }
