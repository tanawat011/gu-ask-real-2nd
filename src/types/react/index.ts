export * from './children'
export * from './reactRouterDom'
