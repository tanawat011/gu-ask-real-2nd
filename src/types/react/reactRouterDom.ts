import type { Scroll } from 'types/common'

export type OutletContextProps = {
  scroll: Scroll
}
