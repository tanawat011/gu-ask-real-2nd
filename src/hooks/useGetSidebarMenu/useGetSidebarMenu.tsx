import type { MenuItem } from 'components/Sidebar'

import type { Dispatch, SetStateAction } from 'react'
import { useState } from 'react'

import { IconHome } from 'components/Icons'
import { mainRoute } from 'routes'

export const useGetSidebarMenu = (): [
  MenuItem[],
  Dispatch<SetStateAction<MenuItem[]>>,
] => {
  const menuSidebar: MenuItem[] = [
    {
      title: ' HOME',
      children: [
        {
          label: 'Dashboard',
          to: mainRoute.home.dashboard,
          icon: <IconHome />,
        },
      ],
    },
    {
      title: 'APP',
      children: [
        {
          label: 'Todo',
          to: mainRoute.app.todo,
          icon: <IconHome />,
        },
      ],
    },
    {
      title: 'GAME',
      children: [
        {
          label: 'Sudoku',
          to: mainRoute.game.sudoku,
          icon: <IconHome />,
        },
        {
          label: 'Tic Tac Toe',
          to: mainRoute.game.ticTacToe,
          icon: <IconHome />,
        },
      ],
    },
    {
      title: 'UI COMPONENT',
      children: [
        {
          label: 'Input',
          icon: <IconHome />,
          children: [
            {
              label: 'Button',
              to: mainRoute.uiComponent.input.button,
            },
            {
              label: 'Button Group',
              to: mainRoute.uiComponent.input.buttonGroup,
            },
            {
              label: 'Calendar',
              to: mainRoute.uiComponent.input.calendar,
            },
            {
              label: 'Checkbox',
              to: mainRoute.uiComponent.input.checkbox,
            },
            {
              label: 'Date And Time',
              to: mainRoute.uiComponent.input.dateAndTime,
            },
            {
              label: 'Radio Button',
              to: mainRoute.uiComponent.input.radioButton,
            },
            {
              label: 'Rating',
              to: mainRoute.uiComponent.input.rating,
            },
            {
              label: 'Select',
              to: mainRoute.uiComponent.input.select,
            },
            {
              label: 'Slider',
              to: mainRoute.uiComponent.input.slider,
            },
            {
              label: 'Switch',
              to: mainRoute.uiComponent.input.switch,
            },
            {
              label: 'Text Field',
              to: mainRoute.uiComponent.input.textField,
            },
          ],
        },
      ],
    },
  ]

  const [menuList, setMenuList] = useState<MenuItem[]>(menuSidebar)

  return [menuList, setMenuList]
}
