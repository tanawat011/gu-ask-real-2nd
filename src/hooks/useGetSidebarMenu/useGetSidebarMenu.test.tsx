import { render } from '@testing-library/react'

import { useGetSidebarMenu } from './useGetSidebarMenu'

jest.mock('utils/route', () => ({
  menuGenerator: () => [
    {
      to: '/todo',
      label: 'Todo',
    },
  ],
}))

const ExampleComponent = () => {
  const [menuList] = useGetSidebarMenu()

  return <div>{menuList[0].title}</div>
}

describe('useGetSidebarMenu', () => {
  test('should return all menu list', () => {
    expect.hasAssertions()

    const { asFragment } = render(<ExampleComponent />)

    expect(asFragment()).toMatchSnapshot()
  })
})
