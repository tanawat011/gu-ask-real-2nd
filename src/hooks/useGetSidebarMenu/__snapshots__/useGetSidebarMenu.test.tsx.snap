// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useGetSidebarMenu should return all menu list 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
